### DISCUSSION TITLE
Feature request: motion reset of sleep timer

### DISCUSSION THREAD

**Discussion by elric667** [TortugaPower/BookPlayer#868] [Ideas]
Sleep timer like the one from smart audiobook player on android

I love their sleep timer function. It’s a unique feature I have never seen in any other audiobook app. You set a time after which the player turns off, but if your device notices movement, the timer resets.

So, if you are still awake and want to continue listening, just give your phone a little shake. No need to look at the screen and reset the sleep timer manually!

**Comment by lexxanite** [comment]
I use AirPods, and if the play button is pressed within such an amount of time (15 seconds ish) then I think it should reset the sleep timer as well. I've gotten used to the smart sleep timer in other apps and think this variant would be an  enhancement on the requested feature as well. No need to reach out to find my phone on my nightstand.

**Comment by GianniCarlo** [comment]
I forgot to follow up here, @elric667 the shake to reset the sleep timer is live now in the current version (I think it was a couple of versions ago)

@lexxanite I was looking through the Airpods controls on [Apple's page](https://support.apple.com/en-vn/guide/airpods/devb2c431317/web), and they just support the regular ones (which they should be already supported in the app):
- 1 press for play/pause
- Double press for next track
- Triple press for previous track

I'm sure there are two more events that involve one tap and hold, and two taps and hold for skip back/forward I believe, not sure why Apple has not listed them for the Airpods though

For reference, the list of remote commands [are here](https://developer.apple.com/documentation/mediaplayer/mpremotecommandcenter), one thing to note, is that adding support for some of them, would alter the way the now-playing widget looks, and only the playback related things are mapped to the earbuds remote commands

**Reply by lexxanite** [reply]
I guess more of the feature I was looking for is just that if play resumes via play button (which I think my single press on AirPods is simply doing) within x amount of time of the sleep timer fading out, reset timer or extend by an amount in settings.

Now that I know shake to extend is in function, that works for me. The request I made is an enhancement on my preferences to interface with the extension more than anything. I just wanna keep my arms in my warm cocoon at night.

**Reply by GianniCarlo** [reply]
ah yes, the single press on the AirPods should already restart the same length or type of timer that just finished (the window should be around 1 min, but could be less due to Apple suspending the app in the background because there's no ongoing playback). I have in mind to do a sort of sticky timer, so a specific timer can be set anytime the user resumes playback, but I have to rework the sleep timer UI and I haven't found time for it yet :/
